src/mon/OSDMonitor.cc

```diff
@@ -395,14 +405,22 @@ void OSDMonitor::update_from_paxos(bool *need_bootstrap)
       t->erase("mkfs", "osdmap");
     }
 
+    // make sure we're using the right pg service.. remove me post-luminous!
+    if (osdmap.require_osd_release >= CEPH_RELEASE_LUMINOUS) {
+      dout(10) << __func__ << " pgservice is mgrstat" << dendl;
+      mon->pgservice = mon->mgrstatmon()->get_pg_stat_service();
+    } else {
+      dout(10) << __func__ << " pgservice is pg" << dendl;
+      mon->pgservice = mon->pgmon()->get_pg_stat_service();
+    }
+
     if (tx_size > g_conf->mon_sync_max_payload_size*2) {
       mon->store->apply_transaction(t);
       t = MonitorDBStore::TransactionRef();
       tx_size = 0;
     }
     if (mon->monmap->get_required_features().contains_all(
           ceph::features::mon::FEATURE_LUMINOUS)) {
-      creating_pgs = update_pending_pgs(inc);
```

Comment:
@gregsfortytwo i think we can drop this line.

if we just deleted a pool with creating pgs, the quorum should be aware of this and `creating_pgs` should be updated. so it should not contain the creatings pgs in that pool. if a new mon just joins the quorum, it will be sync'ed with another monitor. `OSDMonitor::get_store_prefixes()` returns both "osdmap" and "osd_pg_creating", so "pg_creating" will be sync'ed. this ensures us that the new joiner will also have an updated copy of `creating_pgs`.